src/test-integration/java/io/github/jopenlibs/vault/v1_11_4/api/LeasesTests.java

```diff
@@ -40,29 +41,96 @@ public class LeasesTests {
     @BeforeClass
     public static void setupClass() throws IOException, InterruptedException {
         container.initAndUnsealVault();
+        container.setupBackendDatabase(DbContainer.hostname);
     }
 
     @Before
     public void setup() throws VaultException {
         vault = container.getRootVault();
     }
 
-    @Test
+    @Test(expected = VaultException.class)
```

Comment:
Very strange test case. We expect VaultException but the last statement doesn't throw any exception. Please use the `Assert.assertThrows` to specify the statement (/statements) that must be throw an exception. 

Here and below.

Reply:
This case is a a false friend. Let me to explain. To test leases I've choose to use database secret, because it permits to create dynamic credentials. Creation of these credentials releases a lease_id, managed with these methods. Setup of the database secret needs to insert the database commands used to initialize the dynamic user. So, as database engine, I've choose PostgreSQL and I've inserted the `CREATE USER "{{name}}" WITH PASSWORD '{{password}}'` and the `GRANT ALL PRIVILEGES ON DATABASE "postgres" to "{{name}}"` statements, as showed in the code. When I've launched tests, `vault.leases().revoke(lease_id)` method contacted the Vault's REST /sys/leases/revoke endopint, which instructs to send to PostgreSQL a DROP USER <dynamic-name> statement, but PostgreSQL replied with this error:

```
[docker-java-stream--2107830113] INFO io.github.jopenlibs.vault.v1_11_4.util.DbContainer - STDERR: 2023-02-11 18:58:15.179 UTC [109] ERROR:  role "v-root-new-role-VhaQ2FeQiIZwDmnCAUbB-1676141890" cannot be dropped because some objects depend on it
[docker-java-stream-90458516] INFO io.github.jopenlibs.vault.v1_11_4.util.VaultContainer - STDERR: 2023-02-11T18:58:15.179Z [ERROR] secrets.system.system_9ef041bd: revoke prefix failed: prefix=database/creds/new-role error="failed to revoke \"database/creds/new-role/2H16a30vKI16LH7jHAPBQfBu\" (1 / 2): failed to revoke entry: resp: (*logical.Response)(nil) err: ERROR: role \"v-root-new-role-VhaQ2FeQiIZwDmnCAUbB-1676141890\" cannot be dropped because some objects depend on it (SQLSTATE 2BP01)"
[docker-java-stream--2107830113] INFO io.github.jopenlibs.vault.v1_11_4.util.DbContainer - STDERR: 2023-02-11 18:58:15.179 UTC [109] DETAIL:  privileges for database postgres
[docker-java-stream--2107830113] INFO io.github.jopenlibs.vault.v1_11_4.util.DbContainer - STDERR: 2023-02-11 18:58:15.179 UTC [109] STATEMENT:  DROP ROLE IF EXISTS "v-root-new-role-VhaQ2FeQiIZwDmnCAUbB-1676141890";
```

Error in PostgreSQL is caused due to the privileges granted, and in facts is not related how is implemented the functionality or an eventually bug in Vault. For this reason I've used the `@Test(expected = VaultException.class)` annotation for revoke methods, because returned error is not related with the functionality. To correct the tests, I've changed the GRANT statement with one that a less impact, so the annotation isn't necessary